Refactored explorer package

eswarasai opened the pull request:
**What does this pull request do? Explain your changes. (required)**
Updated `components` in `explorer` package to not have greater than 250 lines of code and also added `props` directory to define `type` for individual components

**Specific updates (required)**
- Added `props` directory to define `type` for individual components
- Removed unused variables

**How did you test each of these updates (required)**
NA

**Does this pull request close any open issues?**
Fixes: #87

**Checklist:**
- [x] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

eswarasai commented:
@jozanza -- I haven't refactored the `BasicNavbar` component. It's the only file that's exceeding 250 lines of code in the `explorer` package. And also I haven't created `styles` directory and exporting styles from there as it might be a bit tedious in terms of development to navigate between multiple files in order to update the styles. Let me know your feedback here so that I can update the PR accordingly. Thanks!

jozanza commented:
@eswarasai Thanks for taking the time to do this! Overall, it looks really good. This definitely reduces files size and lines of code 👍 

The only thing to change is rather than having a top-level `props` directory, I think it would be best to actually nest a props file  within its respective component directory. For example:

```
/components
  |- /Button
    |- index.js
    |- props.js
```

This helps keep them self-contained and is especially important since many of these components will eventually move into the [`@livepeer/chroma`](https://github.com/livepeer/livepeerjs/tree/master/packages/chroma) project.

Can you make this update?

jozanza commented:
@eswarasai  Yeah, I'd follow the same approach for the views. They are pretty much set up like this already, so feel free to extract and add a `props.js` in those dirs if desired. But I'd probably do that in a separate PR so this one doesn't get too big :)

eswarasai commented:
@jozanza -- Cool. I'll create a separate PR for the `props` of the `views` 👍

jozanza commented:
Hey @eswarasai be sure to rebase whenever you get a chance. There have been a TON of changes merged recently 🎉

eswarasai commented:
@jozanza -- I've refactored the `props` as per the comments above. Also I've fixed some of the console warnings in the `src/components` for `explorer` package. Please do review the changes and let me know if there's any issue or any other changes required. Thanks!

Also, I'll create a separate PR for the refactor of `views` and also will fix all the console warnings accordingly.

eswarasai commented:
@j0sh @ericxtang @dob -- If someone from Dev team can review this PR and let me know what's pending on this, so that we can get this merged, that'd be great. Thanks!

dob commented:
@eswarasai sorry for the delay in getting this reviewed. We're focused on getting a release out the door this week, and then we can get back to you on next steps here.

eswarasai commented:
@Randy1Burrell -- Sorry for the delay in getting back and thanks for taking care of the feedback as well. Would you need me to handle similar refactor for the `views` as well? This PR just took care of the `components`.

Randy1Burrell commented:
@eswarasai Great work taking care of refactoring the package. I understand how busy and hectic software development gets and sometimes notifications are not seen and responded to immediately. I will create an issue for the view and let you know.
